File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteMiddle = function(head) {
    if ( !head.next){
        return null
    }
        
    let slow = head
    let fast = head
    let temp = null
        
    while (fast && fast.next){
        temp = slow
        fast = fast.next.next
        slow = slow.next
    }
   
    temp.next = slow.next
    slow.next = null
    
    return head
};

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteMiddle(self, head):
        if not head.next:
            return None
        slow = fast = head
        temp = None
        
        while fast and fast.next:
            temp = slow
            fast = fast.next.next
            slow = slow.next
        
        temp.next = slow.next
        slow.next = None
    
        return head
        